unreal/src/error.rs

```diff
@@ -1,36 +1,30 @@
-use failure::Fail;
+use thiserror::Error;
 
 /// Errors related to parsing an UE4 crash file.
 #[non_exhaustive]
-#[derive(Fail, Debug)]
+#[derive(Error, Debug)]
 pub enum Unreal4Error {
     /// Empty data blob received.
-    #[fail(display = "empty crash")]
+    #[error("empty crash")]
     Empty,
 
     /// Invalid compressed data.
-    #[fail(display = "bad compression")]
-    BadCompression(std::io::Error),
+    #[error("bad compression")]
+    BadCompression(#[source] std::io::Error),
 
     /// Invalid contents of the crash file container.
-    #[fail(display = "invalid crash file contents")]
-    BadData(scroll::Error),
+    #[error("invalid crash file contents")]
+    BadData(#[from] scroll::Error),
 
     /// The crash file contains unexpected trailing data after the footer.
-    #[fail(display = "unexpected trailing data")]
+    #[error("unexpected trailing data")]
     TrailingData,
 
     /// Can't process a log entry.
-    #[fail(display = "invalid log entry")]
-    InvalidLogEntry(std::str::Utf8Error),
+    #[error("invalid log entry")]
+    InvalidLogEntry(#[from] std::str::Utf8Error),
```

Comment:
This should not be `#[from]`, as the underlying error is too generic. I believe the same counts for `BadCompression`.

It's too easy where we add a new variant that also contains a `Utf8Error` and then we have to break the `From` impl.